### PR TITLE
Complete BookEditScreen and SelectionManager mapping

### PR DESCRIPTION
**Pull request by yyny**
I'm not entirely happy with some of the mappings I came up with but this should be good enough for documentation purposes.

### CODE REVIEW

File: mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping [hunk 1]
```diff
@@ -44,24 +49,104 @@ CLASS net/minecraft/class_473 net/minecraft/client/gui/screen/ingame/BookEditScr
...
 	METHOD method_27577 invalidatePageContent ()V
 	METHOD method_27578 createPageContent ()Lnet/minecraft/class_473$class_5233;
+	METHOD method_27580 moveCursorLine (I)V
+		ARG 1 offset
+	METHOD method_27581 drawCursor (Lnet/minecraft/class_4587;Lnet/minecraft/class_473$class_5234;Z)V
+		ARG 1 matrices
+		ARG 2 position
+		ARG 3 haveSelection
+	METHOD method_27582 screenToBook (Lnet/minecraft/class_473$class_5234;)Lnet/minecraft/class_473$class_5234;
+		COMMENT Convert a position as a screen coordinate to a position relative to the book UI
```
Comment (enbrain) [inline mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping]:
```suggestion
		COMMENT Converts a position as a screen coordinate to a position relative to the book UI
```

File: mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping [hunk 2]
```diff
@@ -44,24 +49,104 @@ CLASS net/minecraft/class_473 net/minecraft/client/gui/screen/ingame/BookEditScr
...
+		ARG 5 lineTexts
+		ARG 6 style
+		ARG 7 start
+		ARG 8 end
+	METHOD method_27588 drawSelection ([Lnet/minecraft/class_768;)V
+		ARG 1 selectionBounds
+	METHOD method_27589 selectWord (I)V
+		ARG 1 cursorOffset
+	METHOD method_27590 bookToScreen (Lnet/minecraft/class_473$class_5234;)Lnet/minecraft/class_473$class_5234;
+		COMMENT Convert a position relative to the book UI to a position as a screen coordinate
```
Comment (enbrain) [inline mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping]:
```suggestion
		COMMENT Converts a position relative to the book UI to a position as a screen coordinate
```

**Review by modmuss50 — APPROVED** [review]

File: mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping [hunk 3]
```diff
@@ -44,24 +49,104 @@ CLASS net/minecraft/class_473 net/minecraft/client/gui/screen/ingame/BookEditScr
...
+	METHOD method_27584 setClipboard (Ljava/lang/String;)V
+		ARG 1 text
+	METHOD method_27585 createSelectionBoundingBox (Ljava/lang/String;Lnet/minecraft/class_5225;IIII)Lnet/minecraft/class_768;
+		ARG 1 text
+		ARG 2 textHandler
+		ARG 3 fromOffset
+		ARG 4 toOffset
+		ARG 5 lineY
+		ARG 6 lineStart
+	METHOD method_27586 syntheticSplitLine (Lorg/apache/commons/lang3/mutable/MutableInt;Ljava/lang/String;Lorg/apache/commons/lang3/mutable/MutableBoolean;Lit/unimi/dsi/fastutil/ints/IntList;Ljava/util/List;Lnet/minecraft/class_2583;II)V
```
Comment (modmuss50) [inline mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping]: synthetic methods should not have names. IDK how you named these?

**Review by modmuss50 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping (hunk 3 above)
Comment (yyny) [reply mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping]: I'm pretty sure they showed up in Enigma?

**Comment by enbrain** [comment]
You didn't have to remove parameter names from synthetic methods. Only the method names (`syntheticXXX`) were problematic.

File: mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping (hunk 3 above)
Comment (modmuss50) [reply mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping]: Ill take a look. It should only allow you to name the params.

**Review by liach — APPROVED** [review]
Please update pr to merge into 21w03a

**Comment by liach** [comment]
Re synthetic methods:
These methods here are generated by javac because the nested class is calling private members, such as fields and methods, of the enclosing class. Imo you can just name these methods and decompilers will just handle them fine. In original code they are usually named as `$access$00` and one method is generated for each call site in the inner class I think (than one for each field/method in the enclosing class)

**Comment by YanisBft** [comment]
Has been superseded

**Pull request closed without merging**